File: crates/varde-code/tests/fixtures/dart/control_error.dart
void process(int n) {
  if (n > 0) {
    for (var i = 0; i < n; i++) {
      continue;
    }
  }
  while (n > 0) {
    break;
  }
  switch (n) {
    case 1:
      return;
  }
  try {
    throw Exception("boom");
  } on FormatException catch (e) {
    print(e);
  } catch (err) {
    rethrow;
  }
}

File: crates/varde-code/tests/fixtures/javascript/control_error.js
// Control-flow / error fixtures: ControlFlow, Catch, Throw.

function process(items) {
  try {
    if (items.length === 0) {
      throw new Error("empty");
    }
    for (const item of items) {
      while (item > 10) {
        item = item - 1;
        break;
      }
    }
  } catch (err) {
    return `failed: ${err}`;
  } finally {
    cleanup();
  }
}

switch (mode) {
  case "a":
    break;
  default:
    do {
      step();
    } while (running);
}
